core/src/main/scala/akka/persistence/cassandra/query/EventsByPersistenceIdStage.scala

```diff
@@ -320,14 +347,26 @@ import com.datastax.driver.core.utils.Bytes
                 // skip event due to fast forward
                 tryPushOne()
               } else if (config.gapFreeSequenceNumbers && seqNr != event.sequenceNr) {
-                failStage(new IllegalStateException(s"Missing sequence number [$seqNr], got [${event.sequenceNr}] for " +
-                  s"persistenceId [$persistenceId]."))
+                lookingForMissingSeqNr match {
+                  case Some(_) => throw new IllegalStateException(
+                    s"Should not be able to get here when already looking for missing seqNr [$seqNr] for entity [$persistenceId]"
```

Comment:
what is preventing that? let's say we got 2 and 4. 3 is missing, and in next query when looking for 3 we get 5.

Reply:
When we see the gap, we start doing queries on exactly that `seqNr`, and do not toggle back to "normal" querying until that is found and then the `lookingForMissingSeqNr` is reset to `None` below.

Reply:
(so that means we either get 3 or nothing when 3 is missing)

Reply:
ah, I missed that we are only looking for one single event. ok, it's probably good to limit the result set in that way.